candy-machine/program/src/processor/update.rs

```diff
@@ -51,17 +63,18 @@ pub fn handle_update_candy_machine(
     }
 
     let old_uuid = candy_machine.data.uuid.clone();
-    candy_machine.wallet = ctx.accounts.wallet.key();
+    if is_feature_active(&old_uuid, FREEZE_FEATURE_INDEX) && candy_machine.token_mint != token_mint
```

Comment:
It shouldn't actually need to, since a non-withdrawn candy machine account will be passed in to the unfreeze ix. Initially I duplicated the wallet in the freezePDA but I dont think thats needed anymore